lib/knapsack_pro/runners/queue/rspec_runner.rb

```diff
@@ -3,224 +3,283 @@
 module KnapsackPro
   module Runners
     module Queue
+      # Imperative Shell [instance of KnapsackPro::Runners::Queue::RSpecRunner]
+      # Architecture based on: https://www.destroyallsoftware.com/talks/boundaries
+      #
+      # It should contain calls to RSpec methods.
+      # Preferably via methods with the  `knapsack__` prefix.
+      # We want to isolate the imperative shell from the internals of RSpec.
+      # It should be tested via integration tests if possible.
+      # Alternatively, it should be tested using E2E tests on CI that are running against a Rails app with the RSpec test suite.
       class RSpecRunner < BaseRunner
-        @@used_seed = nil
+        # Functional Core.
+        # Architecture based on: https://www.destroyallsoftware.com/talks/boundaries
+        #
+        # It has business logic related to how Knapsack Pro works.
+        # It should be easy to unit test it.
+        # It should NOT contain direct calls to RSpec.
+        class FunctionalCore
```

Comment:
```suggestion
        class Pure
```

Reply:
Also, if we really want to respect the talk Boundaries, we should not inject stuff, but base the communication on data structures.

Instead of injecting a logger and then call `@logger.info` inside the core, we should have the core format the message, return a string, and perform the side effect (logger.info) from the shell.

Reply:
> Instead of injecting a logger and then call @logger.info inside the core, we should have the core format the message, return a string, and perform the side effect (logger.info) from the shell.

Done.